fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/transactionprocessor/impl/AdvancedPaymentScheduleTransactionProcessor.java

```diff
@@ -176,6 +178,153 @@ private void handleRepayment(LoanTransaction loanTransaction, MonetaryCurrency c
             loanTransaction.resetDerivedComponents();
         }
         Money transactionAmountUnprocessed = loanTransaction.getAmount(currency);
+        handleOverAmount(loanTransaction, currency, installments, transactionAmountUnprocessed, charges);
```

Comment:
Please rename this method hence it is not always about to handle overamount... a processTransaction would make more sense to me...